test/pouch_cli_create_test.go

```diff
@@ -1,94 +1,97 @@
-	cmd = PouchCmd{
-		args:        args,
-		result:      true,
-		outContains: "foo",
+	res.Assert(c, icmd.Success)
+	if out := res.Combined(); !strings.Contains(out, name) {
+		c.Fatalf("unexpected output %s expected %s\n", out, name)
 	}
-	RunCmd(c, &cmd)
 }
 
-// TestPouchCreateDuplicateContainerName is to verify duplicate container names.
-func (suite *PouchCreateSuite) TestPouchCreateDuplicateContainerName(c *check.C) {
-	containername := "duplicate"
-	args := []string{"create", "--name", containername, testImage}
+// TestCreateDuplicateContainerName is to verify duplicate container names.
+func (suite *PouchCreateSuite) TestCreateDuplicateContainerName(c *check.C) {
+	name := "duplicate"
 
-	var cmd PouchCmd
+	res := command.PouchRun("create", "--name", name, busyboxImage)
+	res.Assert(c, icmd.Success)
 
-	cmd = PouchCmd{
-		args:        args,
-		result:      true,
-		outContains: containername,
-	}
-	RunCmd(c, &cmd)
+	res = command.PouchRun("create", "--name", name, busyboxImage)
+	c.Assert(res.Error, check.NotNil)
 
-	cmd = PouchCmd{
-		args:        args,
-		result:      false,
-		returnValue: 1,
-		outContains: "already exist",
+	if out := res.Combined(); !strings.Contains(out, "already exist") {
+		c.Fatalf("unexpected output %s expected already exist\n", out)
 	}
-	RunCmd(c, &cmd)
 }
 
-// TestCreateWorks tests "pouch create" work.
-func (suite *PouchCreateSuite) TestCreateWorks(c *check.C) {
+// TestCreateWithArgs is to verify args.
+//
+// TODO: pouch inspect should return args info
+func (suite *PouchCreateSuite) TestCreateWithArgs(c *check.C) {
+	res := command.PouchRun("create", busyboxImage, "/bin/ls")
+	res.Assert(c, icmd.Success)
+}
 
-	args := []string{"create", testImage, "-t"}
-	cmd := PouchCmd{
-		args:   args,
-		result: true,
-	}
-	RunCmd(c, &cmd)
+// TestCreateWithTTY is to verify tty flag.
+//
+// TODO: pouch inspect should return tty info
+func (suite *PouchCreateSuite) TestCreateWithTTY(c *check.C) {
+	res := command.PouchRun("create", "-t", busyboxImage)
+	res.Assert(c, icmd.Success)
+}
 
-	args = []string{"create", "-v", "/tmp:/tmp", testImage}
-	cmd = PouchCmd{
-		args:   args,
-		result: true,
-	}
-	RunCmd(c, &cmd)
+// TestPouchCreateVolume is to verify volume flag.
+//
+// TODO: pouch inspect should return volume info to check
+func (suite *PouchCreateSuite) TestPouchCreateVolume(c *check.C) {
+	res := command.PouchRun("create", "-v /tmp:/tmp", busyboxImage)
+	res.Assert(c, icmd.Success)
+}
 
-	// TODO: clean the created container
+// TestCreateInWrongWay tries to run create in wrong way.
+func (suite *PouchCreateSuite) TestCreateInWrongWay(c *check.C) {
+	for _, tc := range []struct {
+		name string
+		args string
+	}{
+		{name: "unknown flag", args: "-a"},
+
+		// TODO: should add the following cases if ready
+		// {name: "missing image name", args: ""},
+	} {
+		res := command.PouchRun("create", tc.args)
+		c.Assert(res.Error, check.NotNil, check.Commentf(tc.name))
```

Comment:
Yes, I think we should use more `c.Assert` in test cases.  It is much better now. 👍